In [2]:
import pandas as pd
from sqlalchemy import create_engine

DB_CONNECTION = 'postgresql://postgres:postgrespassword@localhost:5432/postgresdb'

engine = create_engine(DB_CONNECTION)

In [3]:
query = "SELECT * FROM analytics_gold.fct_trips LIMIT 5;"
df = pd.read_sql(query, engine)

print("Conexion Exitosa")
display(df.head())

Conexion Exitosa


trip_id     pickup_datetime    dropoff_datetime  \
0  4cb82dab5d54c3977cee3ade150fd0e9 2024-01-15 18:01:11 2024-01-15 19:01:26   
1  b87bea1df0f993cc7c09b56b062010de 2024-01-15 18:01:52 2024-01-15 20:01:14   
2  fe582c58efa267e73eca680e894d7243 2024-01-15 18:01:58 2024-01-15 19:01:23   
3  406154feb6d4e91814f6a9b0e30cca3c 2024-01-15 19:01:00 2024-01-15 20:01:44   
4  c4c550805c96b1d6596cba0717c6a936 2024-01-16 07:01:08 2024-01-16 08:01:10   

   pickup_zone_key  dropoff_zone_key  payment_type_id  trip_distance  \
0              265               216                0          13.72   
1              265                91                0          12.40   
2              265                76                0          18.59   
3              265               232                0           9.33   
4              265               140                0           5.92   

   fare_amount  tip_amount  tolls_amount  total_amount  \
0        42.05         0.0           0.0         42.85   
1        46.79         0.0           0.0         47.59   
2        61.52         0.0           0.0         62.32   
3        35.26         0.0           0.0         36.06   
4        31.06         0.0           0.0         31.86   

                   ingest_ts service_type source_month  
0 2026-02-18 03:22:24.004866       yellow      2024-01  
1 2026-02-18 03:22:24.004866       yellow      2024-01  
2 2026-02-18 03:22:24.004866       yellow      2024-01  
3 2026-02-18 03:22:24.004866       yellow      2024-01  
4 2026-02-18 03:22:24.004866       yellow      2024-01

# Demanda / estacionalidad


1. Viajes por mes(2024)

Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        COUNT(*) as total_viajes
    FROM analytics_gold.fct_trips
    WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime < '2025-01-01'
    GROUP BY 1
    ORDER BY 1;


Tabla usada: analytics_gold.fct_trips

In [3]:
df = pd.read_sql(
    """SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        COUNT(*) as total_viajes
    FROM analytics_gold.fct_trips
    WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime < '2025-01-01'
    GROUP BY 1
    ORDER BY 1;""", engine)
display(df)

mes  total_viajes
0   2024-01       2985425
1   2024-02       3024844
2   2024-03       3595432
3   2024-04       3526724
4   2024-05       3735585
5   2024-06       3544917
6   2024-07       3078130
7   2024-08       2977863
8   2024-09       3631129
9   2024-10       3828307
10  2024-11       3636810
11  2024-12       3651561

Interpretación: El mes con mayor número de viajes es Octubre con 3828307 lo cual puede verse relacionado con los eventos que ocurren en la ciudad para esas fechas (Halloween, Columbus Day Parade, etc) dando una narrativa que la gente suele movilizarse mucho más por estas fechas. En el mes de agosto se registran el menor número de viajes con 2977863 siendo el mes donde se registra el fenómeno conocido como Summer Slump donde la mayoría de locales prefiere retirarse hacia lugares vacacionales. 

2. Viajes por service_type y mes.

Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        service_type,
        COUNT(*) as total_viajes
    FROM analytics_gold.fct_trips
    WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 1, 2;




Tabla usada: analytics_gold.fct_trips


In [4]:
df = pd.read_sql(
    """SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        service_type,
        COUNT(*) as total_viajes
    FROM analytics_gold.fct_trips
    WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 1, 2;""", engine)
display(df)

mes service_type  total_viajes
0   2024-01        green         56370
1   2024-01       yellow       2929055
2   2024-02        green         53403
3   2024-02       yellow       2971441
4   2024-03        green         57254
5   2024-03       yellow       3538178
6   2024-04        green         56256
7   2024-04       yellow       3470468
8   2024-05        green         60780
9   2024-05       yellow       3674805
10  2024-06        green         54556
11  2024-06       yellow       3490361
12  2024-07        green         51641
13  2024-07       yellow       3026489
14  2024-08        green         51622
15  2024-08       yellow       2926241
16  2024-09        green         54261
17  2024-09       yellow       3576868
18  2024-10        green         55999
19  2024-10       yellow       3772308
20  2024-11        green         52067
21  2024-11       yellow       3584743
22  2024-12        green         53812
23  2024-12       yellow       3597749

Interpretación: En general, existe una hegemonía absoluta por parte de los taxis amarillos. Esto refleja correctamente la demanda de taxis con la densidad económica debido a que Manhattan se presenta como un lugar donde existen mayores puestos de trabajo junto con actividades turísticas. Mientras que los taxis verdes se encuentran restringidos hacia las afueras de la ciudad donde se encuentra una mayor cantidad de zonas residenciales. Siendo estas aquellas que prefieren usar transporte propio, Metro o incluso apps como Uber/Lyft. 

3. Top 10 zonas de pickup (total 2024).


 
Query:


    SELECT 
        z.borough,
        z.zone,
        COUNT(*) as total_recogidas
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.pickup_datetime >= '2024-01-01' AND f.pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 3 DESC
    LIMIT 10;




Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [5]:
df = pd.read_sql(
    """SELECT 
        z.borough,
        z.zone,
        COUNT(*) as total_recogidas
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.pickup_datetime >= '2024-01-01' AND f.pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 3 DESC
    LIMIT 10;""", engine)
display(df)

borough                          zone  total_recogidas
0     Queens                   JFK Airport          1910238
1  Manhattan         Upper East Side South          1892766
2  Manhattan                Midtown Center          1886880
3  Manhattan         Upper East Side North          1718374
4  Manhattan                  Midtown East          1399890
5  Manhattan     Times Sq/Theatre District          1362177
6  Manhattan  Penn Station/Madison Sq West          1340583
7  Manhattan           Lincoln Square East          1299330
8     Queens             LaGuardia Airport          1275001
9  Manhattan                   Murray Hill          1160464

Interpretación: El aeropuerto JFK lidera el ranking de pickups reflejando el gran turismo que recibe la ciudad así como también se posiciona como una de las principales sedes a nivel mundial para negocios. A nivel macro, Manhattan sigue siendo uno de los principales lugares de donde se recoge personas formando una triangulación de la demanda en tres tipos de zonas: Hubs de Transporte, Distritos Corporativos y Residencias de Altos Ingresos. 

4. Top 10 zonas de dropoff (total 2024).


Query:


    SELECT 
        z.borough,
        z.zone,
        COUNT(*) as total_bajadas
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.dropoff_zone_key = z.zone_key
    WHERE f.pickup_datetime >= '2024-01-01' AND f.pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 3 DESC
    LIMIT 10;




Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone

In [4]:
df = pd.read_sql(
    """SELECT 
        z.borough,
        z.zone,
        COUNT(*) as total_bajadas
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.dropoff_zone_key = z.zone_key
    WHERE f.pickup_datetime >= '2024-01-01' AND f.pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
    ORDER BY 3 DESC
    LIMIT 10;""", engine)
display(df)

borough                       zone  total_bajadas
0  Manhattan      Upper East Side North        1807546
1  Manhattan      Upper East Side South        1714788
2  Manhattan             Midtown Center        1521470
3  Manhattan  Times Sq/Theatre District        1286408
4  Manhattan                Murray Hill        1199477
5  Manhattan               Midtown East        1171126
6  Manhattan        Lincoln Square East        1137195
7  Manhattan      Upper West Side South        1135541
8  Manhattan               East Chelsea        1063507
9  Manhattan            Lenox Hill West        1058184

Interpretación: Se produce un cambio con respecto a recogidas, ahora podemos observar que las zonas residenciales lideran el ranking de destinos desplazando a las oficinas y al aeropuerto JFK a desaparecer del top. Podemos relacionar esto al comportamiento de "retorno al hogar" donde las personas, de mayor poder adquisitivo, priorizan el taxi como medio de transporte. También podemos observar que lugares turísticos también se encuentran en nuestro ranking, confirmando el hecho de que turistas transitan, en gran cantidad, en taxis

5. Top 5 boroughs por mes (pickup).


Query:


    WITH ranking AS (
        SELECT 
            TO_CHAR(f.pickup_datetime, 'YYYY-MM') as mes,
            z.borough,
            COUNT(*) as viajes,
            RANK() OVER (PARTITION BY TO_CHAR(f.pickup_datetime, 'YYYY-MM') ORDER BY COUNT(*) DESC) as rank
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE f.pickup_datetime >= '2024-01-01'
        GROUP BY 1, 2
    )
    SELECT * FROM ranking WHERE rank <= 5 ORDER BY mes, rank;




Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone

In [14]:
pd.set_option('display.max_rows', None)

In [7]:
df = pd.read_sql(
    """WITH ranking AS (
    SELECT 
        TO_CHAR(f.pickup_datetime, 'YYYY-MM') as mes,
        z.borough,
        COUNT(*) as viajes,
        RANK() OVER (PARTITION BY TO_CHAR(f.pickup_datetime, 'YYYY-MM') ORDER BY COUNT(*) DESC) as rank
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.pickup_datetime >= '2024-01-01' AND f.pickup_datetime < '2025-01-01'
    GROUP BY 1, 2
)
SELECT * FROM ranking WHERE rank <= 5 ORDER BY mes, rank;""", engine)
display(df)

mes    borough   viajes  rank
0   2024-01  Manhattan  2651102     1
1   2024-01     Queens   281597     2
2   2024-01   Brooklyn    32709     3
3   2024-01    Unknown    10378     4
4   2024-01      Bronx     7719     5
5   2024-02  Manhattan  2712717     1
6   2024-02     Queens   257742     2
7   2024-02   Brooklyn    35308     3
8   2024-02    Unknown     9695     4
9   2024-02      Bronx     7641     5
10  2024-03  Manhattan  3196193     1
11  2024-03     Queens   321736     2
12  2024-03   Brooklyn    51767     3
13  2024-03      Bronx    11780     4
14  2024-03    Unknown    11591     5
15  2024-04  Manhattan  3126803     1
16  2024-04     Queens   322499     2
17  2024-04   Brooklyn    52612     3
18  2024-04      Bronx    11475     4
19  2024-04    Unknown    10885     5
20  2024-05  Manhattan  3293132     1
21  2024-05     Queens   353246     2
22  2024-05   Brooklyn    61973     3
23  2024-05      Bronx    12695     4
24  2024-05    Unknown    11696     5
25  2024-06  Manhattan  3121391     1
26  2024-06     Queens   332939     2
27  2024-06   Brooklyn    64591     3
28  2024-06      Bronx    11905     4
29  2024-06    Unknown    11508     5
30  2024-07  Manhattan  2672041     1
31  2024-07     Queens   331352     2
32  2024-07   Brooklyn    52110     3
33  2024-07      Bronx    10336     4
34  2024-07    Unknown     9586     5
35  2024-08  Manhattan  2563262     1
36  2024-08     Queens   339309     2
37  2024-08   Brooklyn    53541     3
38  2024-08      Bronx    10698     4
39  2024-08    Unknown     8282     5
40  2024-09  Manhattan  3167725     1
41  2024-09     Queens   365836     2
42  2024-09   Brooklyn    72432     3
43  2024-09      Bronx    12727     4
44  2024-09    Unknown     9475     5
45  2024-10  Manhattan  3363248     1
46  2024-10     Queens   369693     2
47  2024-10   Brooklyn    69000     3
48  2024-10      Bronx    13211     4
49  2024-10    Unknown    10207     5
50  2024-11  Manhattan  3214690     1
51  2024-11     Queens   324245     2
52  2024-11   Brooklyn    72970     3
53  2024-11      Bronx    13029     4
54  2024-11    Unknown     9431     5
55  2024-12  Manhattan  3227623     1
56  2024-12     Queens   331610     2
57  2024-12   Brooklyn    68173     3
58  2024-12      Bronx    11733     4
59  2024-12    Unknown     9571     5

Interpretación: Se puede observar como existe una centralización del servicio en Manhattan superando a su seguidor más cercano por 10x aprox. Otros distritos presentes se consideran como insignificantes debido a la gran diferencia que hay entre Manhattan y el resto. Esto confirma la naturaleza de los taxis amarillos dentro de NYC donde están diseñados para "parar el taxi levantando la mano", conducta exclusiva de la densidad peatonal de Manhattan. El tráfico de otros barrios se ve proveniente de aeropuertos pudiendo llegar a la conclusión que transitar por estos lugares puede significar una pérdida económica para el taxi.

6. Horas pico (top 5 horas) para cada día de semana.


Query:


    WITH horas_top AS (
        SELECT 
            TO_CHAR(pickup_datetime, 'Day') as dia_nombre,
            EXTRACT(DOW FROM pickup_datetime) as dia_num,
            EXTRACT(HOUR FROM pickup_datetime) as hora,
            COUNT(*) as viajes,
            RANK() OVER (PARTITION BY TO_CHAR(pickup_datetime, 'Day') ORDER BY COUNT(*) DESC) as ranking
        FROM analytics_gold.fct_trips
        GROUP BY 1, 2, 3
    )
    SELECT dia_nombre, hora, viajes 
    FROM horas_top 
    WHERE ranking <= 5 
    ORDER BY dia_num, ranking;




Tabla usada: analytics_gold.fct_trips


In [4]:
df = pd.read_sql(
    """WITH horas_top AS (
            SELECT 
                TO_CHAR(pickup_datetime, 'Day') as dia_nombre,
                EXTRACT(DOW FROM pickup_datetime) as dia_num,
                EXTRACT(HOUR FROM pickup_datetime) as hora,
                COUNT(*) as viajes,
                RANK() OVER (PARTITION BY TO_CHAR(pickup_datetime, 'Day') ORDER BY COUNT(*) DESC) as ranking
            FROM analytics_gold.fct_trips
            GROUP BY 1, 2, 3
        )
        SELECT dia_nombre, hora, viajes 
        FROM horas_top 
        WHERE ranking <= 5 
        ORDER BY dia_num, ranking;""", engine)
display(df)

dia_nombre  hora   viajes
0   Sunday     14.0   955470
1   Sunday     17.0   946067
2   Sunday     16.0   943274
3   Sunday     15.0   931484
4   Sunday     13.0   925056
5   Monday     18.0  1115967
6   Monday     17.0  1091928
7   Monday     15.0  1015917
8   Monday     16.0  1003254
9   Monday     14.0   974424
10  Tuesday    18.0  1316174
11  Tuesday    17.0  1230678
12  Tuesday    19.0  1124126
13  Tuesday    16.0  1097696
14  Tuesday    15.0  1095213
15  Wednesday  18.0  1351324
16  Wednesday  17.0  1263851
17  Wednesday  19.0  1194541
18  Wednesday  16.0  1119124
19  Wednesday  21.0  1112956
20  Thursday   18.0  1383281
21  Thursday   17.0  1285702
22  Thursday   19.0  1216260
23  Thursday   16.0  1143686
24  Thursday   15.0  1137621
25  Friday     18.0  1307242
26  Friday     17.0  1218202
27  Friday     19.0  1212851
28  Friday     15.0  1108914
29  Friday     16.0  1096741
30  Saturday   18.0  1153059
31  Saturday   17.0  1121522
32  Saturday   19.0  1119397
33  Saturday   16.0  1077833
34  Saturday   15.0  1042724

Interpretación: Los datos muestran una mayor cantidad de viajes dentro de la franja de la tarde-noche durante días laborables donde podemos destacar el jueves a las 18h00 como el punto de saturación más alto de toda la semana. Podemos conocer este patrón como "Evening Rush Hour" donde, a diferencia de las mañanas, el regreso a casa desde diferentes distritos genera una demanda explosiva de taxis sugiriendo que el día jueves, con mayor afluencia, se correlaciona también con actividades sociales previa al cierre de la semana laboral. Dentro de los fines de semana, el domingo podemos ver una afluencia mayor a las 14h00 relacionada mayormente con el ocio. 

7. Distribución de viajes por día de semana (ranking).


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'Day') as dia_semana,
        COUNT(*) as total_viajes,
        ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) as porcentaje
    FROM analytics_gold.fct_trips
    GROUP BY 1, EXTRACT(DOW FROM pickup_datetime)
    ORDER BY EXTRACT(DOW FROM pickup_datetime);


Tabla usada: analytics_gold.fct_trips



In [5]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'Day') as dia_semana,
            COUNT(*) as total_viajes,
            ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) as porcentaje
        FROM analytics_gold.fct_trips
        GROUP BY 1, EXTRACT(DOW FROM pickup_datetime)
        ORDER BY EXTRACT(DOW FROM pickup_datetime);""", engine)
display(df)

dia_semana  total_viajes  porcentaje
0  Sunday         14990173       12.48
1  Monday         15148983       12.61
2  Tuesday        17384690       14.47
3  Wednesday      18114575       15.08
4  Thursday       18741650       15.60
5  Friday         18033109       15.01
6  Saturday       17744098       14.77

Interpretación: El volumen de viajes alcanzan su punto máximo el jueves observando un incremento constante desde el lunes hasta el jueves, seguidos de una ligera disminución hacia el fin de semana. Esta tendencia confirma que los taxis en NYC están impulsados por actividades económicas y sociales. El pico del jueves se relaciona con el cierre de la semana laboral junto con actividades recreativas, mientras que el domingo refleja el cese de la mayoría de operaciones. 

# Ingresos / tarifas / propinas

### 8. Ingreso total (total_amount) por mes.


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        SUM(total_amount) as ingresos_totales
    FROM analytics_gold.fct_trips
    GROUP BY 1
    ORDER BY 1;


Tabla usada: analytics_gold.fct_trips


In [7]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
            SUM(total_amount) as ingresos_totales
        FROM analytics_gold.fct_trips
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 1;""", engine)
display(df)

mes  ingresos_totales
0   2024-01      8.161707e+07
1   2024-02      8.220738e+07
2   2024-03      9.965064e+07
3   2024-04      9.912935e+07
4   2024-05      1.085225e+08
5   2024-06      1.015709e+08
6   2024-07      8.918054e+07
7   2024-08      8.715686e+07
8   2024-09      1.067665e+08
9   2024-10      1.121498e+08
10  2024-11      1.037586e+08
11  2024-12      1.071764e+08

Interpretación: Podemos ver la mayor cantidad de dinero recaudado en octubre seguido de mayo mientras que los de menor demanda se registran en enero y febrero. Esto responde, en el caso de la mayor recaudación, a turismo, graduaciones y una mayor actividad por convenciones de negocios. Mientras que podemos ver que en enero y febrero se presenta un clima extremo seguido de la "cuesta de enero" post-festividades reduciendo los viajes cortos como largos.

### 9. Ingreso total por service_type y mes


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        service_type,
        SUM(total_amount) as ingresos
    FROM analytics_gold.fct_trips
    GROUP BY 1, 2
    ORDER BY 1, 2;


Tabla usada: analytics_gold.fct_trips

In [9]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
            service_type,
            SUM(total_amount) as ingresos
        FROM analytics_gold.fct_trips
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1, 2
        ORDER BY 1, 2;""", engine)
display(df)

mes service_type      ingresos
0   2024-01        green  1.268964e+06
1   2024-01       yellow  8.034811e+07
2   2024-02        green  1.214659e+06
3   2024-02       yellow  8.099272e+07
4   2024-03        green  1.318139e+06
5   2024-03       yellow  9.833251e+07
6   2024-04        green  1.322408e+06
7   2024-04       yellow  9.780694e+07
8   2024-05        green  1.501840e+06
9   2024-05       yellow  1.070206e+08
10  2024-06        green  1.363607e+06
11  2024-06       yellow  1.002073e+08
12  2024-07        green  1.285849e+06
13  2024-07       yellow  8.789469e+07
14  2024-08        green  1.348385e+06
15  2024-08       yellow  8.580847e+07
16  2024-09        green  1.454635e+06
17  2024-09       yellow  1.053119e+08
18  2024-10        green  1.407562e+06
19  2024-10       yellow  1.107423e+08
20  2024-11        green  1.264782e+06
21  2024-11       yellow  1.024938e+08
22  2024-12        green  1.296963e+06
23  2024-12       yellow  1.058794e+08

Interpretación: Observamos que los taxis amarillos facturan aproximadamente el 98.5% de la cuota de mercado de términos de ingreso. Se puede notar la correlación con el pico de volúmenes de viajes y la facturación donde: en octubre, siendo el más rentable del año para taxis amarillos coincidiendo con la temporada de convenciones y turismo de otoño, el "valle" del verano, validando la caída de ingresos por el "summer slump". Dentro de estos datos, podemos observar que los taxis verdes presentan valores más bajos pero constantes sugiriendo una demanda residencial inelástica que no depende de turismo o negocios.

### 10. tip % promedio por mes (avg(tip_amount / nullif(fare_amount,0))).


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        AVG(tip_amount / NULLIF(fare_amount, 0)) * 100 as tip_percentage_avg
    FROM analytics_gold.fct_trips
    WHERE fare_amount > 0
    GROUP BY 1
    ORDER BY 1;


Tabla usada: analytics_gold.fct_trips


In [11]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
            AVG(tip_amount / NULLIF(fare_amount, 0)) * 100 as tip_percentage_avg
        FROM analytics_gold.fct_trips
        WHERE fare_amount > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 1;""", engine)
display(df)

mes  tip_percentage_avg
0   2024-01           22.960647
1   2024-02           20.919392
2   2024-03           19.634927
3   2024-04           20.049759
4   2024-05           20.156446
5   2024-06           19.152876
6   2024-07           19.417242
7   2024-08           18.816581
8   2024-09           18.909720
9   2024-10           19.628468
10  2024-11           21.134661
11  2024-12           19.774692

Interpretación: El porcentaje de propina se mantiene alto y estable. Tenemos un dato sorprendente, la mayor cantidad de propinas se recibe en enero con casi un 23% por todo el año mientras que, en agosto, coincide el punto más bajo con el mes de menor actividad. Tener un promedio de 20% no es casualidad, la mayoría de pantallas de tips en NYC ofrecen propinas predeterminadas de entre 20%, 25% y 30% siendo la opción por defecto 20%. En agosto, la conducta de tener menor cantidad de tips puede venir del turismo debido a que vienen personas de otros continentes donde la cultura de propina es diferente o inexistente. 

### 11. tip % por borough y mes.


Query:


    SELECT 
        TO_CHAR(f.pickup_datetime, 'YYYY-MM') as mes,
        z.borough,
        AVG(f.tip_amount / NULLIF(f.fare_amount, 0)) * 100 as tip_pct
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.fare_amount > 0
    GROUP BY 1, 2
    ORDER BY 1, 2;


Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [15]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(f.pickup_datetime, 'YYYY-MM') as mes,
            z.borough,
            AVG(f.tip_amount / NULLIF(f.fare_amount, 0)) * 100 as tip_pct
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE f.fare_amount > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1, 2
        ORDER BY 1, 2;""", engine)
display(df)

mes        borough      tip_pct
0   2024-01          Bronx     2.466335
1   2024-01       Brooklyn     9.401475
2   2024-01            EWR    47.763597
3   2024-01      Manhattan    21.708639
4   2024-01         Queens    20.965924
5   2024-01  Staten Island     6.230469
6   2024-01        Unknown    20.687266
7   2024-01            NaN  2938.170899
8   2024-02          Bronx     2.888597
9   2024-02       Brooklyn    11.952097
10  2024-02            EWR    35.767823
11  2024-02      Manhattan    21.055074
12  2024-02         Queens    17.646580
13  2024-02  Staten Island     5.830027
14  2024-02        Unknown    21.092133
15  2024-02            NaN   687.828589
16  2024-03          Bronx    17.680255
17  2024-03       Brooklyn    21.568772
18  2024-03            EWR    64.966495
19  2024-03      Manhattan    19.889448
20  2024-03         Queens    16.733155
21  2024-03  Staten Island     9.155149
22  2024-03        Unknown    20.249103
23  2024-03            NaN    32.424899
24  2024-04          Bronx     2.133063
25  2024-04       Brooklyn     9.233497
26  2024-04            EWR    29.356290
27  2024-04      Manhattan    20.072402
28  2024-04         Queens    16.714898
29  2024-04  Staten Island    27.444498
30  2024-04        Unknown    20.760851
31  2024-04            NaN   983.451934
32  2024-05          Bronx     2.307701
33  2024-05       Brooklyn     6.790023
34  2024-05            EWR    38.381265
35  2024-05      Manhattan    19.998897
36  2024-05         Queens    16.206294
37  2024-05  Staten Island     3.916200
38  2024-05        Unknown    20.437083
39  2024-05            NaN  1452.942630
40  2024-06          Bronx    56.162238
41  2024-06       Brooklyn     6.592249
42  2024-06            EWR    18.981575
43  2024-06      Manhattan    19.622557
44  2024-06         Queens    15.803310
45  2024-06  Staten Island     2.628969
46  2024-06        Unknown    20.255124
47  2024-06            NaN    24.951570
48  2024-07          Bronx     4.175879
49  2024-07       Brooklyn     6.713057
50  2024-07            EWR    14.670492
51  2024-07      Manhattan    19.836697
52  2024-07         Queens    18.328264
53  2024-07  Staten Island     2.574868
54  2024-07        Unknown    19.949144
55  2024-07            NaN    48.873037
56  2024-08          Bronx     2.330300
57  2024-08       Brooklyn     6.938311
58  2024-08            EWR    13.527350
59  2024-08      Manhattan    19.575655
60  2024-08         Queens    15.360368
61  2024-08  Staten Island    28.346584
62  2024-08        Unknown    19.753237
63  2024-08            NaN    35.716058
64  2024-09          Bronx     2.745656
65  2024-09       Brooklyn    12.071131
66  2024-09            EWR    44.081248
67  2024-09      Manhattan    19.059685
68  2024-09         Queens    15.579806
69  2024-09  Staten Island     1.253358
70  2024-09        Unknown    20.349440
71  2024-09            NaN   674.171897
72  2024-10          Bronx     2.309306
73  2024-10       Brooklyn     6.552917
74  2024-10            EWR    54.029035
75  2024-10      Manhattan    19.798265
76  2024-10         Queens    18.401306
77  2024-10  Staten Island     1.077689
78  2024-10        Unknown    20.280541
79  2024-10            NaN   475.487985
80  2024-11          Bronx     2.025686
81  2024-11       Brooklyn     5.215853
82  2024-11            EWR  1917.842346
83  2024-11      Manhattan    21.269233
84  2024-11         Queens    20.254840
85  2024-11  Staten Island     2.978627
86  2024-11        Unknown    20.069000
87  2024-11            NaN   174.412800
88  2024-12          Bronx     2.461530
89  2024-12       Brooklyn     6.273462
90  2024-12            EWR    19.814160
91  2024-12      Manhattan    20.497158
92  2024-12         Queens    16.070358
93  2024-12  Staten Island    25.045407
94  2024-12        Unknown    19.861719
95  2024-12            NaN    26.169792

Interpretación: Los datos muestran que Manhattan mantiene un promedio de propina estable durante todo el año, seguido por Queens. Podemos observar que otros distritos muestran volatilidad extrema siendo síntoma de volumen de datos estadisticamente insuficiente. Dentro del dataset pudimos encontrar anomalías en la zona EWR y en los registros nulos, donde podemos encontrar valores superiores a 1000% pudiendo clasificarlos como ruido que no refleja el actuar de los usuarios.

### 12. Top 10 zonas por ingreso total (pickup).


Query:


    SELECT 
        z.zone,
        SUM(f.total_amount) as ingreso_total
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    GROUP BY 1
    ORDER BY 2 DESC
    LIMIT 10;


Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone

Interpretación:

In [20]:
df = pd.read_sql(
    """SELECT 
            z.zone,
            SUM(f.total_amount) as ingreso_total
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE f.fare_amount > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 2 DESC
        LIMIT 10;""", engine)
display(df)

zone  ingreso_total
0                   JFK Airport   4.312875e+08
1             LaGuardia Airport   2.241207e+08
2                Midtown Center   1.215488e+08
3         Upper East Side South   1.057665e+08
4     Times Sq/Theatre District   1.021162e+08
5         Upper East Side North   9.783966e+07
6  Penn Station/Madison Sq West   9.150421e+07
7                  Midtown East   9.103673e+07
8           Lincoln Square East   7.934141e+07
9                 Midtown North   7.816887e+07

Interpretación: Dentro de nuestro dataset podemos encontrar que el aeropuerto JFK genera la mayor cantidad de dinero, doblando a su seguidor más cercano. Aunque, anteriormente, se observó que tenian un volumen de viaje casi similar con Midtown Center, JFK genera casi 4 veces más ingresos donde podemos concluir que los viajes en Midtown Center son viajes cortos, atrapados en el tráfico con una tarifa baja mientras que los viajes de aeropuerto son de larga distancia.

### 13. Top 10 zonas por tip % (pickup) con mínimo N viajes (define N y documenta).


Query:


    SELECT 
        z.zone,
        COUNT(*) as total_viajes,
        AVG(f.tip_amount / NULLIF(f.fare_amount, 0)) * 100 as avg_tip_pct
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.fare_amount > 0
    GROUP BY 1
    HAVING COUNT(*) >= 50  -- Definimos N=50 para evitar outliers estadísticos
    ORDER BY 3 DESC
    LIMIT 10;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [21]:
df = pd.read_sql(
    """SELECT 
            z.zone,
            COUNT(*) as total_viajes,
            AVG(f.tip_amount / NULLIF(f.fare_amount, 0)) * 100 as avg_tip_pct
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE f.fare_amount > 5 AND f.fare_amount > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        HAVING COUNT(*) >= 50  -- Definimos N=50 para evitar outliers estadísticos
        ORDER BY 3 DESC
        LIMIT 10;""", engine)
display(df)

zone  total_viajes  avg_tip_pct
0  Upper East Side South       1848943    21.752081
1  Upper East Side North       1668245    21.313478
2  Upper West Side South       1098753    21.177530
3      LaGuardia Airport       1268926    21.120966
4    Lincoln Square East       1272733    21.118551
5           Midtown East       1377737    20.880560
6        Lenox Hill West        890858    20.597531
7               Union Sq       1071354    20.533890
8               Flatiron        626003    20.510225
9          Midtown North       1133673    20.364989

Interpretación: En el siguiente análisis se tuvo que colocar como fare_amount valores mayores a 5 para poder trabajar con datos reales, puesto que anteriormente se presentaban propinas astronómicas de 544% su valor, esto siendo resultado de tarifas negociadas o viajes fuera de la jurisdicción estandar donde el conductor puede llegar a negociar un precio fijo que se ingresa como extra dejando el valor fare_amount muy bajo. 
Para proceder con el análisis, después de eliminar los valores astronómicos contamos con que las 10 zonas más "generosas" se encuentran dentro de Manhattan manteniendo una propina promedio casi idéntica. Este top 10 está compuesto por barrios residenciales de alta renta y distritos de entretenimiento premium donde la cultura de propina del 20% es un estándar social entre residentes.

### 14. Comparación cash vs card: viajes, ingreso total, tip %.


Query:


    SELECT 
        p.payment_type_desc,
        COUNT(*) as viajes,
        SUM(total_amount) as ingreso_total,
        AVG(tip_amount / NULLIF(fare_amount, 0)) * 100 as avg_tip_pct
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_payment_type p ON f.payment_type_id = p.payment_type_id
    WHERE f.fare_amount > 0
    GROUP BY 1;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_payment_type


In [ ]:
df = pd.read_sql(
    """SELECT 
            p.payment_type_desc,
            COUNT(*) as viajes,
            SUM(total_amount) as ingreso_total,
            AVG(tip_amount / NULLIF(fare_amount, 0)) * 100 as avg_tip_pct
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_payment_type p ON f.payment_type_id = p.payment_type_id
        WHERE f.fare_amount > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1;""", engine)
display(df)

payment_type_desc    viajes  ingreso_total  avg_tip_pct
0              Cash   5563144   1.378993e+08     0.001436
1       Credit card  30904403   9.223756e+08    26.136024
2           Dispute    423712   1.200243e+07     0.085763
3             Empty   3959570   1.003646e+08     3.559069
4         No charge    202140   5.092586e+06     0.124847
5           Unknown        29   6.408400e+02     0.000000

Interpretación: El uso de Credit Card es el estandar absoluto. Podemos identificar una brecha crítica en el registro de propinas, donde las tarjetas reportan un 26.13% de propiina promedio, mientras que el efectivo registra un 0.001% donde podemos concluir que esto no implica que el usuario no deje propina, si no que al ser propinas "físicas" no pasan por el taximetro quedando fuera de la trazabilidad del dato.

# Duración / distancia / eficiencia

### 15. Duración promedio (min) por mes.


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        AVG(EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as duracion_promedio_min
    FROM analytics_gold.fct_trips
    GROUP BY 1
    ORDER BY 1;

Tabla usada: analytics_gold.fct_trips

Interpretación:

In [26]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
            AVG(EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as duracion_promedio_min
        FROM analytics_gold.fct_trips
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 1;""", engine)
display(df)

mes  duracion_promedio_min
0   2024-01              15.705130
1   2024-02              16.068035
2   2024-03              16.734817
3   2024-04              17.105001
4   2024-05              18.101222
5   2024-06              17.642570
6   2024-07              17.286845
7   2024-08              17.449333
8   2024-09              18.692736
9   2024-10              18.343675
10  2024-11              17.900442
11  2024-12              18.767719

Interpretación: Podemos ver un aumento progresivo en el tiempo de viaje a lo largo del año. Iniciamos con enero donde tenemos la duración más baja, esto puede deberse a que existe un menor volumen tras las fiestas, mientras que para Septiembre a Diciembre nos encontramos con el regreso a clases, aumento de eventos y el turismo navideño y festividades.

### 16. Distancia promedio por mes


Query:


    SELECT 
        TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
        AVG(trip_distance) as distancia_avg_millas
    FROM analytics_gold.fct_trips
    GROUP BY 1
    ORDER BY 1;

Tabla usada: analytics_gold.fct_trips


In [27]:
df = pd.read_sql(
    """SELECT 
            TO_CHAR(pickup_datetime, 'YYYY-MM') as mes,
            AVG(trip_distance) as distancia_avg_millas
        FROM analytics_gold.fct_trips
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 1;""", engine)
display(df)

mes  distancia_avg_millas
0   2024-01              4.186956
1   2024-02              4.115020
2   2024-03              4.676128
3   2024-04              5.415806
4   2024-05              5.544082
5   2024-06              5.380554
6   2024-07              5.237334
7   2024-08              5.216556
8   2024-09              5.963982
9   2024-10              5.240001
10  2024-11              5.684178
11  2024-12              5.241745

Interpretación: Podemos observar un patron en los primeros meses del año (enero-marzo) donde se realizan mayor cantidad de viajes cortos debido al frío de NYC reduciendo el promedio de millas. A partir de abril las distancias se estabilizan por encima de las 5.2 millas teniendo como pico máxico Septiembre donde se refleja el inicio de clases, eventos masivos y la semana de la moda. 

### 17. Velocidad promedio (mph) por borough y franja horaria.


Query:


    SELECT 
        z.borough,
        EXTRACT(HOUR FROM f.pickup_datetime) as hora_dia,
        AVG(f.trip_distance / NULLIF(EXTRACT(EPOCH FROM (f.dropoff_datetime - f.pickup_datetime))/3600, 0)) as velocidad_promedio_mph
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    WHERE f.trip_distance > 0 
    AND f.dropoff_datetime > f.pickup_datetime
    GROUP BY 1, 2
    ORDER BY 1, 2;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [28]:
df = pd.read_sql(
    """SELECT 
            z.borough,
            EXTRACT(HOUR FROM f.pickup_datetime) as hora_dia,
            AVG(f.trip_distance / NULLIF(EXTRACT(EPOCH FROM (f.dropoff_datetime - f.pickup_datetime))/3600, 0)) as velocidad_promedio_mph
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE f.trip_distance > 0 AND pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        AND f.dropoff_datetime > f.pickup_datetime
        GROUP BY 1, 2
        ORDER BY 1, 2;""", engine)
display(df)

borough  hora_dia  velocidad_promedio_mph
0            Bronx       0.0               22.710514
1            Bronx       1.0               22.388191
2            Bronx       2.0              579.944641
3            Bronx       3.0              162.262662
4            Bronx       4.0               20.962078
5            Bronx       5.0               44.984352
6            Bronx       6.0              127.431542
7            Bronx       7.0              276.844710
8            Bronx       8.0              162.888617
9            Bronx       9.0              154.119150
10           Bronx      10.0               83.574004
11           Bronx      11.0               62.158782
12           Bronx      12.0               51.171701
13           Bronx      13.0               14.100464
14           Bronx      14.0              307.898331
15           Bronx      15.0              169.417247
16           Bronx      16.0               17.647326
17           Bronx      17.0               79.591399
18           Bronx      18.0               22.999691
19           Bronx      19.0               84.400700
20           Bronx      20.0               19.507173
21           Bronx      21.0               56.445083
22           Bronx      22.0               43.876835
23           Bronx      23.0               40.636471
24        Brooklyn       0.0               53.598133
25        Brooklyn       1.0               27.196394
26        Brooklyn       2.0               25.404418
27        Brooklyn       3.0               20.504512
28        Brooklyn       4.0              108.557719
29        Brooklyn       5.0               97.766744
30        Brooklyn       6.0               35.418387
31        Brooklyn       7.0              114.927224
32        Brooklyn       8.0              123.001583
33        Brooklyn       9.0               73.948058
34        Brooklyn      10.0               28.961646
35        Brooklyn      11.0               29.962529
36        Brooklyn      12.0               52.037362
37        Brooklyn      13.0               54.973010
38        Brooklyn      14.0               54.172996
39        Brooklyn      15.0               71.327304
40        Brooklyn      16.0               48.038503
41        Brooklyn      17.0              186.519886
42        Brooklyn      18.0               35.283002
43        Brooklyn      19.0               36.932930
44        Brooklyn      20.0               33.854927
45        Brooklyn      21.0               32.555608
46        Brooklyn      22.0               54.980390
47        Brooklyn      23.0               34.496925
48             EWR       0.0               15.718347
49             EWR       1.0               16.661900
50             EWR       2.0               24.888170
51             EWR       3.0               18.932551
52             EWR       4.0               25.244487
53             EWR       5.0               62.902919
54             EWR       6.0               38.285267
55             EWR       7.0              189.984057
56             EWR       8.0               70.879824
57             EWR       9.0             1449.860683
58             EWR      10.0              193.926228
59             EWR      11.0              136.345805
60             EWR      12.0              112.050895
61             EWR      13.0              265.067122
62             EWR      14.0               77.832364
63             EWR      15.0              211.201454
64             EWR      16.0              252.964518
65             EWR      17.0              200.104172
66             EWR      18.0               97.203535
67             EWR      19.0               96.588257
68             EWR      20.0              208.260654
69             EWR      21.0              131.485599
70             EWR      22.0               19.057689
71             EWR      23.0              243.345475
72       Manhattan       0.0               15.373448
73       Manhattan       1.0               16.459904
74       Manhattan       2.0        

Interpretación: Si filtramos errores técnicos podemos observar cómo Manhattan tiene una velocidad promedio de apenas 12 a 15 MPH duante la mayor parte del día siendo un punto de congestión entre las 12h00 y 13h00, así como las 18h00. Ahora, podemos detectar outliers críticos en todos los distritos, esto debido a que el cálculo realizado puede presentar en segundos una duración casi de 0 mientras que en millas una gran cantidad.

### 18. Percentiles p50 y p90 de duración por borough.


Query:


    SELECT 
        z.borough,
        PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p50_duracion_min,
        PERCENTILE_CONT(0.9) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p90_duracion_min
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    GROUP BY 1;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [29]:
df = pd.read_sql(
    """SELECT 
            z.borough,
            PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p50_duracion_min,
            PERCENTILE_CONT(0.9) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p90_duracion_min
        FROM analytics_gold.fct_trips f
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        GROUP BY 1;""", engine)
display(df)

borough  p50_duracion_min  p90_duracion_min
0          Bronx         24.450000         61.966667
1       Brooklyn         20.900000         50.233333
2            EWR          0.150000          1.383333
3      Manhattan         11.733333         26.216667
4         Queens         32.483333         60.633333
5  Staten Island         36.550000         81.113333
6        Unknown         14.133333         38.333333
7            NaN          0.950000         60.333333

Interpretación: Existe una brecha masiva entre Manhattan y distritos exteriores donde se puede llegar a triplicar la duración del viaje teniendo datos extremos como en Staten Island, donde el p90 supera los 81 minutos mientras que Manhattan el peor escenario se situa en los 26. Esto consolida a Manhattan como un hub de micro-movilidad en contraste con otros barrios como Queens y Staten Island donde existen viajes de mayor duración.

### 19. Top 10 zonas (pickup) por p90 de duración.


Query:


    SELECT 
        z.zone,
        PERCENTILE_CONT(0.9) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p90_duracion
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
    GROUP BY 1
    ORDER BY 2 DESC
    LIMIT 10;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone


In [31]:
df = pd.read_sql(
    """SELECT 
            z.zone,
            PERCENTILE_CONT(0.9) WITHIN GROUP (ORDER BY EXTRACT(EPOCH FROM (dropoff_datetime - pickup_datetime))/60) as p90_duracion
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone z ON f.pickup_zone_key = z.zone_key
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1
        ORDER BY 2 DESC
        LIMIT 10;""", engine)
display(df)

zone  p90_duracion
0     Far Rockaway    105.416667
1  Hammels/Arverne    103.878333
2     Coney Island    100.283333
3    Rockaway Park     94.543333
4   Brighton Beach     92.673333
5         Rosedale     89.500000
6        Gravesend     89.210000
7       Co-Op City     88.103333
8  Cambria Heights     88.036667
9        Laurelton     87.310000

Interpretación: Identificamos un grupo de zonas periféricas donde el 10% de los viajes superan la barrera de los 90 a 105 minutos. Todas las zonas comparten la característica de ser puntos geográficamente más alejados del Distrito Financiero y Midtown Manhattan. Toma sentido también que un viaje inicial de Rockaways o Rosedale que tenga como destino Manhattan tenga múltiples cuellos de botella y cruzar distritos enteros que se traduce en que uno de cada diez viajes tome casi dos horas de tiempo.

# Rutas / patrones

### 20. Top 10 rutas borough→borough (pickup borough to dropoff borough) por número de viajes.


Query:


    SELECT 
        zp.borough as origen_borough,
        zd.borough as destino_borough,
        COUNT(*) as total_viajes
    FROM analytics_gold.fct_trips f
    JOIN analytics_gold.dim_zone zp ON f.pickup_zone_key = zp.zone_key
    JOIN analytics_gold.dim_zone zd ON f.dropoff_zone_key = zd.zone_key
    GROUP BY 1, 2
    ORDER BY 3 DESC
    LIMIT 10;

Tabla usada: analytics_gold.fct_trips, analytics_gold.dim_zone

Interpretación:

In [32]:
df = pd.read_sql(
    """SELECT 
            zp.borough as origen_borough,
            zd.borough as destino_borough,
            COUNT(*) as total_viajes
        FROM analytics_gold.fct_trips f
        JOIN analytics_gold.dim_zone zp ON f.pickup_zone_key = zp.zone_key
        JOIN analytics_gold.dim_zone zd ON f.dropoff_zone_key = zd.zone_key
        WHERE pickup_datetime >= '2024-01-01' AND pickup_datetime <= '2025-01-01'
        GROUP BY 1, 2
        ORDER BY 3 DESC
        LIMIT 10;""", engine)
display(df)

origen_borough destino_borough  total_viajes
0      Manhattan       Manhattan      34079843
1         Queens       Manhattan       2140373
2      Manhattan          Queens       1097878
3         Queens          Queens       1020801
4      Manhattan        Brooklyn        751498
5         Queens        Brooklyn        564230
6       Brooklyn        Brooklyn        386478
7       Brooklyn       Manhattan        206411
8      Manhattan           Bronx        126308
9         Queens             NaN        110268

Interpretación: Los datos revelan una monopolización extrema en la ruta de Manhattan, donde Manhattan -> Manhattan supera a la segunda ruta por un factor de 16 a 1. Esto confirma que el taxi no es un transporte de largas distancias, es una herramienta de micro-movilidad interna para el centro de la ciudad. También podemos encontrar el volumen de rutas de Queens -> Manhattan como un flujo constante de turistas que ingresan y salen de la ciudad. 